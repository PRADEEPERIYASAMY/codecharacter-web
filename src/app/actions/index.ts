export * from 'app/actions/dashboard';
