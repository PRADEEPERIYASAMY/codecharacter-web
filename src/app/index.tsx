import { Dashboard } from 'app/components/Dashboard';
import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router';

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <Switch>
    <Route path="/" component={Dashboard} />
  </Switch>
));
