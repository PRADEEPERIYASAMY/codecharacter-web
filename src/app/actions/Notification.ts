import { NotificationTabType, NotificationType } from 'app/types/Notification';
import { action } from 'typesafe-actions';

export namespace NotificationActions {
  export enum Type {
    INFO = 'INFO',
    SUCCESS = 'SUCCESS',
    ERROR = 'ERROR',
    ADD = 'ADD',
    DELETE_NOTIFICATION = 'DELETE_NOTIFICATION',
    DELETE_NOTIFICATION_TYPE = 'DELETE_NOTIFICATION_TYPE',
  }

  export const info = (title: string, text: string) =>
    action(Type.INFO, {
      text,
      title,
    });

  export const success = (title: string, text: string) =>
    action(Type.SUCCESS, {
      text,
      title,
    });

  export const error = (title: string, text: string) =>
    action(Type.ERROR, {
      text,
      title,
    });
  export const add = (type: NotificationType, title: string, text: string) =>
    action(Type.ADD, {
      text,
      title,
      type,
    });

  export const deleteNotificationType = (type: NotificationTabType) =>
    action(Type.DELETE_NOTIFICATION_TYPE, {
      type,
    });

  export const deleteNotification = (id: number) =>
    action(Type.DELETE_NOTIFICATION, {
      id,
    });
}
