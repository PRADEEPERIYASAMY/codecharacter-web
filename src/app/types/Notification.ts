import { NotificationActions } from 'app/actions';
import { ActionType } from 'typesafe-actions';

const actions = {
  error: NotificationActions.error,
  info: NotificationActions.info,
  success: NotificationActions.success,
};

export type NotificationStoreAction = ActionType<typeof actions>;
