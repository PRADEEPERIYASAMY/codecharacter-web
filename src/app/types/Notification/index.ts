import { NotificationActions } from 'app/actions';
import { ActionType } from 'typesafe-actions';

const actions = {
  add: NotificationActions.add,
  deleteNotification: NotificationActions.deleteNotification,
  deleteNotificationType: NotificationActions.deleteNotificationType,
  error: NotificationActions.error,
  info: NotificationActions.info,
  success: NotificationActions.success,
};

export enum NotificationType {
  INFO = 'INFO',
  SUCCESS = 'SUCCESS',
  ERROR = 'ERROR',
}

export enum NotificationTabType {
  ALL = 'ALL',
  INFO = 'INFO',
  SUCCESS = 'SUCCESS',
  ERROR = 'ERROR',
}

export interface Notification {
  id: number;
  title: string;
  text: string;
  type: NotificationType;
}

export interface NotificationStoreState {
  loading: boolean;
  notifications: Notification[];
}

export interface State {
  activeNotificationTab: NotificationTabType;
}

export interface StateProps {
  loading: boolean;
  notifications: Notification[];
}

export interface DispatchProps {
  deleteNotificationType: (type: NotificationTabType) => void;
}

export type Props = StateProps & DispatchProps;

export type NotificationStoreAction = ActionType<typeof actions>;
