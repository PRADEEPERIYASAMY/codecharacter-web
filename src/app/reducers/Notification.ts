import { NotificationActions } from 'app/actions';
import * as NotificationInterfaces from 'app/types/Notification';

// @ts-ignore
import PNotify from 'pnotify/dist/es/PNotify';
import 'pnotify/dist/PNotifyBrightTheme.css';

const notificationInitialState: {} = {};

export const notificationReducer = (
  state = notificationInitialState,
  action: NotificationInterfaces.NotificationStoreAction,
) => {
  switch (action.type) {
    case NotificationActions.Type.INFO: {
      PNotify.info({
        text: action.payload.text,
        title: action.payload.title,
      });
      return state;
    }
    case NotificationActions.Type.SUCCESS: {
      PNotify.success({
        text: action.payload.text,
        title: action.payload.title,
      });
      return state;
    }
    case NotificationActions.Type.ERROR: {
      PNotify.error({
        text: action.payload.text,
        title: action.payload.title,
      });
      return state;
    }
    default:
      return state;
  }
};
