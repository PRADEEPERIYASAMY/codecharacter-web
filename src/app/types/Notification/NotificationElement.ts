import { Notification } from 'app/types/Notification';

export interface DispatchProps {
  deleteNotification: (id: number) => void;
}

export type Props = Notification & DispatchProps;
