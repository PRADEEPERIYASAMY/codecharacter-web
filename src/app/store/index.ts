import { logger } from 'app/middleware';
import { rootReducer } from 'app/reducers';
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

export function configureStore(initialState?: object) {
  let middleware = applyMiddleware(logger);

  if (process.env.NODE_ENV !== 'production') {
    middleware = composeWithDevTools(middleware);
  }

  const store = createStore(rootReducer, initialState, middleware);

  if (module.hot) {
    module.hot.accept('app/reducers', () => {
      const nextReducer = require('app/reducers');
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
