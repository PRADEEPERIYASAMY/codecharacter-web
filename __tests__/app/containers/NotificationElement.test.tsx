import NotificationElement from 'app/containers/Notification/NotificationElement';
import { configureStore } from 'app/store';
import * as NotificationInterfaces from 'app/types/Notification';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('Notification Container', () => {
  const props = {
    id: 1,
    text: 'You have moved up a postion in leaderboard',
    title: 'Info!',
    type: NotificationInterfaces.NotificationType.INFO,
  };
  it('Should render Notification', () => {
    const store = configureStore();
    const wrapper = shallow(<NotificationElement {...props} />, {
      context: {
        store,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have props', () => {
    const store = configureStore();
    const wrapper = shallow(<NotificationElement {...props} />, {
      context: {
        store,
      },
    });
    expect(wrapper.props().id).toEqual(props.id);
    expect(wrapper.props().text).toEqual(props.text);
    expect(wrapper.props().title).toEqual(props.title);
    expect(wrapper.props().type).toEqual(props.type);
  });
});
