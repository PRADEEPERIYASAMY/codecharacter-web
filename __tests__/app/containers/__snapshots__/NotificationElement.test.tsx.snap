// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Container Should render Notification 1`] = `
<NotificationElement
  deleteNotification={[Function]}
  id={1}
  text="You have moved up a postion in leaderboard"
  title="Info!"
  type="INFO"
/>
`;
